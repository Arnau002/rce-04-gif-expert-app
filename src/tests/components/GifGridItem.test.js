import { shallow } from "enzyme"
import { GifGridItem } from "../../components/GifGridItem"

describe('Test <GifGridItem />', () => { 

    const title = 'Un título';
    const url = 'https://localhost/img.jpg'

    let wrapper = shallow(<GifGridItem title={title} url={url}/>)

    test('Es Mostra Correctament', () => { 
        expect(wrapper).toMatchSnapshot();
    });

    test('Ha de tenir un paràgraf amb el title', () => { 
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(title);
    });

    test('La imatge ha de tenir src = url i alt = title', () => { 
        const img = wrapper.find('img');
        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);
     });

     test('Té animate_fadeIn', () => { 
        const div = wrapper.find('div');
        const className = div.prop('className');
        expect(className.includes('animate__fadeIn')).toBe(true); 
      });      
 });