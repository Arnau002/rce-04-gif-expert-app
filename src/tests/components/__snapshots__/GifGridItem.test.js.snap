// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <GifGridItem /> Es Mostra Correctament 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Un título"
    src="https://localhost/img.jpg"
  />
  <p>
    Un título
  </p>
</div>
`;
